openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -394,3 +415,37 @@ def get_group_info_for_cohort(cohort):
     if len(res):
         return res[0].group_id, res[0].partition_id
     return None, None
+
+
+def set_assignment_type(user_group, assignment_type):
+    """
+    Set assignment type for cohort.
+    """
+    course_cohort = user_group.cohort
+
+    if is_default_cohort(user_group) and course_cohort.assignment_type != assignment_type:
+        raise ValueError(_("There must be one cohort to which students can be randomly assigned."))
```

Comment:
This exception message will be passed to frontend and user will see this message.